### PR TITLE
[.NET] Disable output embedding on macOS, move it to the advanced options on other platforms.

### PR DESCRIPTION
**Pull request by bruvzg**
- Disable output embedding on macOS (it can't work, and have no benefits since macOS app is a bundle with multiple files anyway).
- Hides `output embedding` options on macOS (always disable), iOS (should use iOS specific code path and ignore it) and Android (always enabled).
- Move `output embedding` to the advanced options on Linux and Windows (while it can be used on these platforms, embedding and extracting native libs seems like an extremely bad idea, I would consider removing this option entirely, but using it at least should not be encouraged).

Note: removed some dead code and added `ifdef` to fix .Net build with `werror=yes` (unrelated to the issue).

Fixes (partially, .Net 8 AOT seems to be broken, but it's likely an unrelated issue) https://github.com/godotengine/godot/issues/90397

### CODE REVIEW

**Comment by bruvzg** [comment]
For the reference, the embedding option was added as part of Android support - https://github.com/godotengine/godot/pull/73257, but it is available on other platforms as well. Enabling it most likely should not cause any issues on Linux, but on Windows it will prevent exporter from signing libraries and might attract unnecessary antivirus software attention. The only benefit seems to be exporting without extra file (apart for executable and .pck).

File: modules/mono/csharp_script.cpp [hunk 1]
```diff
@@ -410,115 +410,6 @@ ScriptLanguage::ScriptNameCasing CSharpLanguage::preferred_file_name_casing() co
 }
 
 #ifdef TOOLS_ENABLED
-struct VariantCsName {
-	Variant::Type variant_type;
-	const String cs_type;
-};
-
-static String variant_type_to_managed_name(const String &p_var_type_name) {
```
Comment (raulsntos) [inline modules/mono/csharp_script.cpp:418]: This became unused with https://github.com/godotengine/godot/pull/87952 and it's OK to remove it.

File: platform/android/export/export_plugin.cpp [hunk 2]
```diff
@@ -1939,6 +1939,12 @@ bool EditorExportPlatformAndroid::get_export_option_visibility(const EditorExpor
 		// The APK templates are ignored if Gradle build is enabled.
 		return advanced_options_enabled && !bool(p_preset->get("gradle_build/use_gradle_build"));
 	}
+
+	// Hide .Net embedding option (always enabled).
```
Comment (raulsntos) [inline platform/android/export/export_plugin.cpp:1943]: Minor nit:

```suggestion
	// Hide .NET embedding option (always enabled).
```

File: platform/ios/export/export_plugin.cpp [hunk 3]
```diff
@@ -135,6 +135,11 @@ void EditorExportPlatformIOS::_notification(int p_what) {
 }
 
 bool EditorExportPlatformIOS::get_export_option_visibility(const EditorExportPreset *p_preset, const String &p_option) const {
+	// Hide unsupported .Net embedding option.
```
Comment (raulsntos) [inline platform/ios/export/export_plugin.cpp:138]:
```suggestion
	// Hide unsupported .NET embedding option.
```

File: platform/macos/export/export_plugin.cpp [hunk 4]
```diff
@@ -333,6 +333,12 @@ bool EditorExportPlatformMacOS::get_export_option_visibility(const EditorExportP
 			return false;
 		}
 	}
+
+	// Hide unsupported .Net embedding option.
```
Comment (raulsntos) [inline platform/macos/export/export_plugin.cpp:337]:
```suggestion
	// Hide unsupported .NET embedding option.
```

**Review by raulsntos — APPROVED** [review]
This looks good to me, although I'm not sure how the platform maintainers feel about adding more .NET-specific code to their areas. I think there's precedent in https://github.com/godotengine/godot/pull/88245 that supports this.

**Comment by akien-mga** [comment]
> This looks good to me, although I'm not sure how the platform maintainers feel about adding more .NET-specific code to their areas. I think there's precedent in #88245 that supports this.

Yeah ideally we'd have this better abstracted, so modules can keep their platform-specific code well encapsulated, and likewise platforms should be able to have some module-specific code well encapsulated.

But I don't have a clear idea for how to do this so for now I think it's fine to just hardcode a few checks. We can refactor down the line once there are more modules (especially third-party) that require this kind of hooks.

File: modules/mono/csharp_script.cpp (hunk 1 above)
Comment (akien-mga) [reply modules/mono/csharp_script.cpp:418]: Thanks for pointing it out, so I know this part of the commit should _not_ be cherry-picked to `4.2`.

**Comment by bruvzg** [comment]
> Yeah ideally we'd have this better abstracted, so modules can keep their platform-specific code well encapsulated, and likewise platforms should be able to have some module-specific code well encapsulated.

Export options will need some refactoring anyway, advanced options support was recently added and seems to be only used for Android, and in https://github.com/godotengine/godot/pull/90375. But it is useful for the all platforms (some export options are rarely used), so some way for modules to control export option visibility can be added (I do not think there's any, and currently only export platform plugins can do it).

In case of this feature, I'm not sure if it should be an option at all (or it should be hardcoded to be used only for Android).

**Comment by akien-mga** [comment]
Thanks!

**Comment by raulsntos** [comment]
> In case of this feature, I'm not sure if it should be an option at all (or it should be hardcoded to be used only for Android).

After working on https://github.com/godotengine/godot/pull/88803 I'm not sure it should be used for Android either. As I understand it, this would include the build outputs for all architectures in the APK in a way that can't be stripped. So users end up downloading a bigger APK that contains all architectures when they only need the files for their target architecture.

This feature was implemented in https://github.com/godotengine/godot/pull/76305. Users were asking to include the exported C# files in the PCK to avoid having an extra directory. This is not relevant on Android where we already only have the APK, but it's still relevant for desktop platforms. @bruvzg Is there a way to include these files inside the PCK without the caveats you mentioned? (i.e.: not signing the libraries on Windows, etc.)

**Comment by bruvzg** [comment]
> Is there a way to include these files inside the PCK without the caveats you mentioned? (i.e.: not signing the libraries on Windows, etc.)

Libs can be signed before packing. But personally, I would prefer to remove it completely, it is a pointless hack, it's leaving garbage in the AppData, and it's potential vulnerability since anything can replace files in AppData.

**Pull request merged**